src/groupeddataframe/grouping.jl

```diff
@@ -14,9 +14,9 @@ view into the AbstractDataFrame grouped by rows.
 
 Not meant to be constructed directly, see `groupby`.
 """
-mutable struct GroupedDataFrame
-    parent::AbstractDataFrame
-    cols::Vector         # columns used for sorting
+mutable struct GroupedDataFrame{T<:AbstractDataFrame}
```

Comment:
What is the benefit of `GroupedDataFrame` being mutable (and does it have performance impact)?

(as a side note - I have just noticed the same question applies to `DataFrame`)

Reply:
Probably none. I'm surprised we haven't noticed this before. Let's change this in another PR.